curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa204c1e1d33348ff09944.md

```diff
@@ -1,17 +1,17 @@
 ---
 id: 62aa204c1e1d33348ff09944
-title: Step 151
+title: Step 150
 challengeType: 0
-dashedName: step-151
+dashedName: step-150
 ---
 
 # --description--
 
-Move your `monsterHealth` assignment into your `if` block.
+In your `attack` function, find the line of code that updates the `monsterHealth` variable and place it within a new `if` block with a condition that calls the `isMonsterHit` function.
 
 # --hints--
 
-You should move your `monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1` into your `if` block.
+You should move your `monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1` into a new `if` block.
 
 ```js
 assert.match(attack.toString(), /if\s*\(\s*isMonsterHit\(\s*\)\s*\)\s*\{\s*monsterHealth\s*-=\s*weapons\s*\[\s*currentWeapon\s*\]\s*\.power\s*\+\s*Math\.floor\(\s*Math\.random\(\s*\)\s*\*\s*xp\s*\)\s*\+\s*1/)
```

Comment:
The tests here will need to be more granular - something like:

- You should have a new `if` block
- Your `if` condition should call the `isMonsterHit` function
- Your `monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1` should be executed when `isMonsterHit()` is true.

Reply:
@naomi-lgbt I was unable to add the You should have a new `if` block test due to there being two other if blocks in the attack function already. I have this test instead:

Your `attack` function should have a new `if` block that calls the `isMonsterHit` function.

I also am not sure how to implement the last test you mentioned:

Your `monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1` should be executed when `isMonsterHit()` is true.

Reply:
>  You should have a new if block test due to there being two other if blocks in the attack function already

Get the number of if blocks with a `String.match(RegExp)` call and assert its length is 3.